rustls/tests/api.rs

```diff
@@ -3986,3 +3989,100 @@ fn test_acceptor() {
         .unwrap();
     assert!(acceptor.accept().is_err());
 }
+
+#[derive(Default, Debug)]
+struct LogCounts {
+    trace: usize,
+    debug: usize,
+    info: usize,
+    warn: usize,
+    error: usize,
+}
+
+impl LogCounts {
+    fn new() -> Self {
+        Self {
+            ..Default::default()
+        }
+    }
+
+    fn reset(&mut self) {
+        *self = Self::new();
+    }
+
+    fn add(&mut self, level: log::Level) {
+        match level {
+            log::Level::Trace => self.trace += 1,
+            log::Level::Debug => self.debug += 1,
+            log::Level::Info => self.info += 1,
+            log::Level::Warn => self.warn += 1,
+            log::Level::Error => self.error += 1,
+        }
+    }
+}
+
+thread_local!(static COUNTS: RefCell<LogCounts> = RefCell::new(LogCounts::new()));
+
+struct CountingLogger;
+
+static LOGGER: CountingLogger = CountingLogger;
+
+impl CountingLogger {
+    fn install() {
+        COUNTS.with(|c| {
+            c.borrow_mut().reset();
+        });
+        log::set_logger(&LOGGER).unwrap();
```

Comment:
This is a bit confusing because the logger can only be set once per process. IMO it would be clearer to have `reset()` that resets the counts and `install()` that does the rest. It doesn't matter now but it will probably matter later.